src/main/java/com/heatheryou/httpserver/RequestParser.java

```diff
@@ -9,18 +10,33 @@
 
 public class RequestParser {
     private List<String> requestList;
-    String uri;
-    String method;
-    String body;
+    private ISystemOutput systemOutput;
 
-    public Request processRequest(BufferedReader requestReader) throws IOException {
+    public RequestParser(ISystemOutput systemOutput) {
```

Comment:
The `RequestParser` seems better to me if it doesn't take the `systemOutput`.  Sometimes adding another dependency is fine or even necessary.  But in this particular case, printing to the system console almost seems like it's taking an another responsibility?  Is there a way to avoid this?

Maybe validate the request first somehow?  Have a `RequestValidator`?  Or have a `validate()` method on the parser?

I'm not sure how this would work out - but maybe reading all the lines from the `BufferedReader` before we ever get to this class, and then using those lines here instead of the `BufferedReader` would facilitate some changes to improve the code?  Maybe a class could exist just to do that if necessary?